src/carbonio-files-ui-common/views/components/List.test.tsx

```diff
@@ -0,0 +1,39 @@
+/*
+ * SPDX-FileCopyrightText: 2023 Zextras <https://www.zextras.com>
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+import React from 'react';
+
+import { screen } from '@testing-library/react';
+
+import { List } from './List';
+import { populateNodes } from '../../mocks/mockUtils';
+import { selectNodes, setup } from '../../utils/testUtils';
+
+describe('List', () => {
+	describe('Badge', () => {
+		test('should render the list header with the badge with the number of selected items', async () => {
+			const nodes = populateNodes(10);
+			const { user } = setup(<List nodes={nodes} mainList emptyListMessage={'hint'} />);
```

Comment:
Even if no graphql mock is required in this test case, the usage of the mocked provider is preferred over the usage of the apollo provider with msw. In order to use the first one, specify the mocks field inside the second arg of the setup function. The value is an empty array to indicate that no query is run across this test.

```suggestion
			const { user } = setup(<List nodes={nodes} mainList emptyListMessage={'hint'} />, { mocks: [] );
```